apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/PluginCfg.java

```diff
@@ -52,14 +52,13 @@ void load(InputStream input) throws IOException {
                     LOGGER.error(e, "Failed to format plugin({}) define.", pluginDefine);
                 }
             }
-            pluginClassList = pluginSelector.select(pluginClassList);
         } finally {
             input.close();
         }
     }
 
     public List<PluginDefine> getPluginClassList() {
-        return pluginClassList;
+        return pluginSelector.select(pluginClassList);
```

Comment:
```suggestion
        pluginClassList = pluginSelector.select(pluginClassList);
        return pluginClassList;
```

You should make sure the excluded ones could be GC, right?

Reply:
Thanks, a new migic I learned.